### PR TITLE
Add MAppbar

### LINKED ISSUE #13: Add AppBar
- Material Design Guideline
  - [App bars : bottom](https://material.io/design/components/app-bars-bottom.html)
  - [App bars : top](https://material.io/design/components/app-bars-top.html)

### PR DESCRIPTION
**Pull request by flyofsky**
### Description of Change ###
This PR is for adding MAppbar

### Issues Resolved ### 
- fixes #13 

### API Changes ###
Added
 - MAppbar

### Platforms Affected ### 
- Tizen

### Behavioral/Visual Changes ###
None

### Before/After Screenshots ### 
 - Pair
Selection_026

 - Top
Selection_021

 - Prominent
Selection_022

 - Bottom
Selection_023

 - FAB Left
Selection_025

 - FAB Right
Selection_024

### PR Checklist ###

- [ ] Has automated tests 
- [x] Rebased on top of the target branch at time of PR
- [x] Changes adhere to coding standard

### CODE REVIEW

File: src/Tizen.NET.MaterialComponents/Components/MAppbar.cs [hunk 1]
```diff
@@ -0,0 +1,261 @@
...
+                    image = null;
+                }
+            }
+        }
+
+        void IColorSchemeComponent.OnColorSchemeChanged(bool fromConstructor)
+        {
+            if (_naviItem != null)
+            {
+                _naviItem.TitleBarBackgroundColor = MColors.Current.PrimaryColor;
```
Comment (rookiejava) [inline src/Tizen.NET.MaterialComponents/Components/MAppbar.cs]: Is there no way to change `TextColor`?

File: src/Tizen.NET.MaterialComponents/Components/MAppbar.cs [hunk 2]
```diff
@@ -0,0 +1,261 @@
...
+
+                button.SetPartContent(Parts.Widget.Icon, image);
+                _naviItem.SetPartContent(position, button);
+            }
+            else
+            {
+                _naviItem.SetPartContent(position, null);
+                if (button != null)
+                {
+                    button = null;
```
Comment (rookiejava) [inline src/Tizen.NET.MaterialComponents/Components/MAppbar.cs]: without Unrealize()?

File: src/Tizen.NET.MaterialComponents/Components/MAppbar.cs [hunk 3]
```diff
@@ -0,0 +1,261 @@
...
+            {
+                _naviItem.SetPartText(Styles.Default, _title);
+            }
+            else
+            {
+                _naviItem.SetPartText(Styles.Default, string.Empty);
+            }
+        }
+
+        void UpdateItem(string position, string item, ref MButton button, ref Image image)
```
Comment (rookiejava) [inline src/Tizen.NET.MaterialComponents/Components/MAppbar.cs]: What about `Button  CreateItem(string icon)` instead?

File: src/Tizen.NET.MaterialComponents/Components/MAppbar.cs [hunk 4]
```diff
@@ -0,0 +1,261 @@
...
+                return _overflowItem;
+            }
+            set
+            {
+                _overflowItem = value;
+                UpdateItem(Parts.NaviItem.OverflowItem, _overflowItem, ref _overflowButton, ref _overflowImage);
+            }
+        }
+
+        public event EventHandler NavigationItemClicked
```
Comment (rookiejava) [inline src/Tizen.NET.MaterialComponents/Components/MAppbar.cs]: It looks like the order has been reversed. 

What about add invoke code insider the button event handler as below?
```cs
public void OnNavigationButtonClicked()
{
    NavigationItemClicked?.Invoke(this, EventArgs.Empty);
}
```

**Review by rookiejava — CHANGES_REQUESTED** [review]
Couldn't we use toolbar for implementing the AppBar? Plus, this seems to satisfy only the regular top appbar, so it seems apporriate to change the name `MAppBar` to `MTopAppBar` or 'MTopRegularAppBar` for the sake of intuitiveness.

File: src/Tizen.NET.MaterialComponents/Components/MAppbar.cs (hunk 1 above)
Comment (flyofsky) [reply src/Tizen.NET.MaterialComponents/Components/MAppbar.cs]: Done

File: src/Tizen.NET.MaterialComponents/Components/MAppbar.cs (hunk 2 above)
Comment (flyofsky) [reply src/Tizen.NET.MaterialComponents/Components/MAppbar.cs]: Done

File: src/Tizen.NET.MaterialComponents/Components/MAppbar.cs (hunk 3 above)
Comment (flyofsky) [reply src/Tizen.NET.MaterialComponents/Components/MAppbar.cs]: Done

File: src/Tizen.NET.MaterialComponents/Components/MAppbar.cs (hunk 4 above)
Comment (flyofsky) [reply src/Tizen.NET.MaterialComponents/Components/MAppbar.cs]: Done

File: src/Tizen.NET.MaterialComponents/ColorExtensions.cs [hunk 5]
```diff
@@ -18,5 +18,9 @@ public static Color WithAlpha(this Color color, double alpha)
             return new Color(color.R, color.G, color.B, (int)(255 * alpha));
         }
 
+        public static string ToHex(Color color)
```
Comment (rookiejava) [inline src/Tizen.NET.MaterialComponents/ColorExtensions.cs]: ToHex(this Color color)

File: src/Tizen.NET.MaterialComponents/Constants.cs [hunk 6]
```diff
@@ -5,6 +5,7 @@ namespace Tizen.NET.MaterialComponents
     public static class Styles
     {
         public static readonly string Material = "material";
+        public static readonly string Default = "default";
```
Comment (rookiejava) [inline src/Tizen.NET.MaterialComponents/Constants.cs]: where it used?

**Review by rookiejava — COMMENTED** [review]
I just looked a quick glance and I have one simple question. Can both TopAppBar and BottomAppBar are existing at once?

**Comment by flyofsky** [comment]
Yes, according to the guide below, TopAppbar and BottomAppbar can be placed at the same time.
https://material.io/design/components/app-bars-bottom.html#behavior
Check Placement part.

File: src/Tizen.NET.MaterialComponents/ColorExtensions.cs (hunk 5 above)
Comment (flyofsky) [reply src/Tizen.NET.MaterialComponents/ColorExtensions.cs]: Done

File: src/Tizen.NET.MaterialComponents/Constants.cs (hunk 6 above)
Comment (flyofsky) [reply src/Tizen.NET.MaterialComponents/Constants.cs]: Done

**Comment by shyunMin** [comment]
New PR(#53 ) has been opened according to design changes.

**Pull request closed without merging**